### PR TITLE
Extension of localLinks structure in io-package.json

### PR DESCRIPTION
**Pull request by GermanBluefox**
I want to use `localLinks` for all types of links: intro, pro, cloud and in instances

### CODE REVIEW

File: schemas/io-package.json [hunk 1]
```diff
@@ -1179,21 +1179,67 @@
             ".+": {
               "anyOf": [
                 {
-                  "type": "string"
+                  "type": "string",
+                  "deprecationMessage": "Please use structure {link, color, pro, cloud, intro, icon, name, order} instead of string"
```
Comment (foxriver76) [inline schemas/io-package.json:1183]: This property is not in the specs. Either write `DEPRECATED:` in the description or simply remove this option so developers will get a warning from the IDE.

File: schemas/io-package.json [hunk 2]
```diff
@@ -1179,21 +1179,67 @@
...
+                      "description": "Link to the web service of this adapter, like: %web_protocol%://%ip%:%web_port%/vis-2/edit.html"
                     },
                     "color": {
                       "type": "string"
                     },
                     "pro": {
+                      "anyOf": [
+                        {
+                          "type": "boolean",
+                          "deprecationMessage": "Please use link template as string instead of boolean"
```
Comment (foxriver76) [inline schemas/io-package.json:1198]: same

File: schemas/io-package.json [hunk 3]
```diff
@@ -1179,21 +1179,67 @@
...
+                          "type": "string"
+                        }
+                      ]
+                    },
+                    "cloud": {
+                      "description": "Link to the adapter, like: vis-2/index.html according to \"https://iobroker.net/\"",
+                      "type": "string"
+                    },
+                    "intro": {
+                      "description": "Should be this link be shown on intro tab in admin",
```
Comment (foxriver76) [inline schemas/io-package.json:1211]:
```suggestion
                      "description": "If this link should be shown on the intro tab in admin",
```

File: schemas/io-package.json [hunk 4]
```diff
@@ -1461,6 +1507,7 @@
           }
         },
         "welcomeScreen": {
+          "deprecationMessage": "Please use common.localLinks instead",
```
Comment (foxriver76) [inline schemas/io-package.json:1510]: not in spec, see recommendations above

File: schemas/io-package.json [hunk 5]
```diff
@@ -1471,6 +1518,7 @@
           }
         },
         "welcomeScreenPro": {
+          "deprecationMessage": "Please use common.localLinks instead",
```
Comment (foxriver76) [inline schemas/io-package.json:1521]: not in spec, see recommendations above

File: packages/types-dev/objects.d.ts [hunk 6]
```diff
@@ -626,25 +626,39 @@ declare global {
...
             /** Which version of this adapter is installed */
             installedVersion: string;
             keywords?: string[];
             /** A dictionary of links to web services this adapter provides */
-            localLinks?: Record<string, string>;
+            localLinks?: Record<
+                string,
+                | string
+                | {
+                      link: string;
```
Comment (foxriver76) [inline packages/types-dev/objects.d.ts:641]: we could add the descriptions from schema also here as JSDOC like comments, so you have this information during development

**Review by foxriver76 — CHANGES_REQUESTED** [review]

File: schemas/io-package.json (hunk 1 above)
Comment (GermanBluefox) [reply schemas/io-package.json]: Done

**Review by foxriver76 — APPROVED** [review]

**Pull request merged**